src/main/java/teammates/ui/webapi/GetNotificationsAction.java

```diff
@@ -63,11 +64,44 @@ public JsonResult execute() {
                     logic.getActiveNotificationsByTargetUser(targetUser);
         }
 
-        NotificationsData responseData = new NotificationsData(notificationAttributes);
+        String isFetchingAllString = getRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL);
+        if (isFetchingAllString != null
+                && !isFetchingAllString.equals("true") && !isFetchingAllString.equals("false")) {
+            throw new InvalidHttpParameterException(INVALID_IS_FETCHING_ALL);
+        }
+        boolean isFetchingAll = Boolean.parseBoolean(isFetchingAllString);
```

Comment:
Could also do this:
```
boolean isFetchingAll = false;
if (getRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL) != null) {
    isFetchingAll = getBooleanRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL);
}
```

Reply:
`getBooleanRequestParamValue` will not throw error if the string is not null, as this change is in response to @NicolasCwy 's suggestion of checking isfetching is not assigned a non boolean value.

Reply:
It looks like `getBooleanRequestParamValue()` will never throw an `IllegalArgumentException` because `Boolean.parseBoolean()` doesn't. In this case you should correct the `getBooleanRequestParamValue()` method instead to throw `IllegalArgumentException` if the values are neither "true" nor "false" (ignoring case).